From 139e01bb5ae3c4cf28722f96796baed6f1ff3b01 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Tue, 27 Jan 2026 10:26:02 +0100
Subject: [PATCH 2/2] Do not run PARI code when cross-compiling

---
 meson.build | 44 +++++++++++++++++++++++---------------------
 1 file changed, 23 insertions(+), 21 deletions(-)

diff --git a/meson.build b/meson.build
index f73517c..9d52155 100644
--- a/meson.build
+++ b/meson.build
@@ -57,27 +57,29 @@ pari = cc.find_library(
 pari = declare_dependency(include_directories: pari_header_include_dirs,
   dependencies: [pari, gmp])
 
-# Get PARI version (mostly as smoke test)
-pari_version_code = '''
-#include <stdio.h>
-#include <pari/pari.h>
-int main(void) {
-  pari_init(1000000, 2);
-  GEN v = pari_version();
-  const char *version = GENtostr(pari_version());
-  pari_printf("%s",version);
-  pari_close();
-  return 0;
-}
-'''
-pari_version = cc.run(
-  pari_version_code,
-  args: ['-v'],
-  name: 'pari version',
-  dependencies: [pari],
-  required: true,
-)
-message('PARI Version: ' + pari_version.stdout())
+if meson.can_run_host_binaries()
+  # Get PARI version (mostly as smoke test)
+  pari_version_code = '''
+  #include <stdio.h>
+  #include <pari/pari.h>
+  int main(void) {
+    pari_init(1000000, 2);
+    GEN v = pari_version();
+    const char *version = GENtostr(pari_version());
+    pari_printf("%s",version);
+    pari_close();
+    return 0;
+  }
+  '''
+  pari_version = cc.run(
+    pari_version_code,
+    args: ['-v'],
+    name: 'pari version',
+    dependencies: [pari],
+    required: true,
+  )
+  message('PARI Version: ' + pari_version.stdout())
+endif
 
 pari_datadir_code = '''
 #include <stdio.h>
-- 
2.52.0

