From f07dd7a4d03dba4a4d6d3e295844aabc88767eef Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Tue, 27 Jan 2026 11:27:18 +0100
Subject: [PATCH 3/3] Allow providing datadir when cross-compiling

---
 meson.build | 38 ++++++++++++++++++++++----------------
 1 file changed, 22 insertions(+), 16 deletions(-)

diff --git a/meson.build b/meson.build
index 9d52155..3b0d4c4 100644
--- a/meson.build
+++ b/meson.build
@@ -81,22 +81,28 @@ if meson.can_run_host_binaries()
   message('PARI Version: ' + pari_version.stdout())
 endif
 
-pari_datadir_code = '''
-#include <stdio.h>
-#include <pari/pari.h>
-int main(void) {
-  pari_init(1000000, 2);
-  pari_printf("%s", pari_datadir);
-  pari_close();
-  return 0;
-}
-'''
-pari_datadir = cc.run(
-  pari_datadir_code,
-  name: 'pari datadir',
-  dependencies: [pari],
-  required: true,
-).stdout()
+pari_share = meson.get_env('PARI_SHARE')
+if pari_share != ''
+  pari_datadir = pari_share
+  message('PARI_SHARE set; using: ' + pari_datadir)
+else
+  pari_datadir_code = '''
+  #include <stdio.h>
+  #include <pari/pari.h>
+  int main(void) {
+    pari_init(1000000, 2);
+    pari_printf("%s", pari_datadir);
+    pari_close();
+    return 0;
+  }
+  '''
+  pari_datadir = cc.run(
+    pari_datadir_code,
+    name: 'pari datadir',
+    dependencies: [pari],
+    required: true,
+  ).stdout()
+endif
 
 # In MSYS, convert to Windows path
 # (e.g /usr/share/pari -> C:\msys64\usr\share\pari)
-- 
2.52.0

