From ede26bedbdd6744f8ca32275774361d0dfd68afa Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Sat, 29 May 2021 11:13:06 -0500
Subject: [PATCH] Use PARI_SHARE env variable to support cross compiling

---
 autogen/paths.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/autogen/paths.py b/autogen/paths.py
index caea02e..3e8194f 100644
--- a/autogen/paths.py
+++ b/autogen/paths.py
@@ -40,6 +40,8 @@ def pari_share():
         >>> os.path.isfile(os.path.join(pari_share(), "pari.desc"))
         True
     """
+    if "PARI_SHARE" in os.environ:
+        return os.environ["PARI_SHARE"]
     from subprocess import Popen, PIPE
     if not gppath:
         raise EnvironmentError("cannot find an installation of PARI/GP: make sure that the 'gp' program is in your $PATH")
